hometask2/hometask2_1.js

```diff
@@ -0,0 +1,145 @@
+const express = require("express");
+const app = express();
+
+const bodyParser = require("body-parser");
+app.use(bodyParser.json() );
+app.use(bodyParser.urlencoded({
+  extended: true
+})); 
+
+
+const Joi = require("@hapi/joi");
+const validator = require("express-joi-validation").createValidator({});
+
+const schema = Joi.object({
+	login: Joi.string()
+		.alphanum()
+		.min(5)
+		.max(30)
+		.required(),
+
+	password: Joi.string()
+		.pattern(new RegExp("^[a-zA-Z0-9]{8,30}$"))
+		.required(),
+
+	age: Joi.number()
+		.integer()
+		.min(4)
+		.max(130)
+		.required(),
+});
+
+
+
+const usersMap = new Map();
+let uid = 1;
+
+//read
+app.get("/user/:id", (req, res) => {
+	
+	let id = req.params.id;
+	
+	let user = usersMap.get(id);
+	if (user && !user.isDeleted) {
+		res.send("user = " + JSON.stringify(user));
+	} else {
+		res.send("user not found, id = " + id);
+	}
+	
+});
+
+//autosuggest
+app.get("/users/:login/:limit", (req, res) => {
+	
+	const login = req.params.login;
+	let limit = parseInt(req.params.limit);
+
+	let count = 0;
+	let userResult = new Array();
+	const pattern = RegExp(".*" + login + ".*");
+	for (const [id, user] of usersMap) {
+		if (pattern.test(user.login) && !user.isDeleted) {
+			userResult.push(user);
+			count++;
+			if (count === limit) {
+				break;
+			}
+		}
+	}
+
+	if (userResult.length > 0) {
+		userResult.sort((a, b) => {
+			var loginA = a.login.toUpperCase();
+			var loginB = b.login.toUpperCase();
+	
+			if (loginA < loginB) {
+				return -1;
+			}
+			if (loginA > loginB) {
+				return 1;
+			}
+	
+			return 0;
+		});
+	
+		res.send("user = " + JSON.stringify(userResult));
+	} else {
+		res.send("users not found");
+	}
+});
+
+//create
+app.post("/user", validator.body(schema), (req, res) => {
+	let id = uid.toString();
+
+	let user = {
```

Comment:
Don't forget to use "const" for everything that doesn't change in the current scope.